test/machines/utils.go

```diff
@@ -0,0 +1,42 @@
+package machines
+
+import (
+	. "github.com/onsi/gomega"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clusterv1alpha1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+
+	"github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/framework"
+
+	providerconfigv1 "sigs.k8s.io/cluster-api-provider-aws/pkg/apis/awsproviderconfig/v1alpha1"
+	machineutils "sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/actuators/machine"
+)
+
+func createSecretAndWait(f *framework.Framework, secret *apiv1.Secret) {
+	_, err := f.KubeClient.CoreV1().Secrets(secret.Namespace).Create(secret)
+	Expect(err).NotTo(HaveOccurred())
+
+	err = wait.Poll(framework.PollInterval, framework.PoolTimeout, func() (bool, error) {
+		if _, err := f.KubeClient.CoreV1().Secrets(secret.Namespace).Get(secret.Name, metav1.GetOptions{}); err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func getMachineStatus(framework *framework.Framework, machine *clusterv1alpha1.Machine) string {
```

Comment:
This is called `getMachineStatus` but is returning a condition of the `machine.Status.ProviderStatus`
We should have two helpers:
- `getMachineStatus`, returns `Machine.Status`
For this we want to verify:
`Machine.Status.Addresses`
`Machine.Status.Conditions`
- `getMachineProviderStatus`, returns `Machine.Status.ProviderStatus`:
For this we want to verify:
`ProviderStatus.Conditions`